libr/anal/fcn.c

```diff
@@ -1818,14 +1822,22 @@ R_API int r_anal_fcn_add_bb(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut64 siz
 			r_anal_fcn_update_tinyrange_bbs (fcn);
 		}
 	}
-	if (!bb) {
-		bb = appendBasicBlock (anal, fcn, addr);
-		if (!bb) {
-			eprintf ("appendBasicBlock failed\n");
-			return false;
-		}
+	if (bb) {
+		r_list_delete_data (fcn->bbs, bb);
+	}
+	old_num_bbs = r_list_length (fcn->bbs);
+	bbuf = malloc (MAXBBSIZE);
+	if (bbuf) {
+		anal->iob.read_at (anal->iob.io, addr, bbuf, MAXBBSIZE);
+		fcn_recurse (anal, fcn, addr, bbuf, MAXBBSIZE, 1);
+		r_anal_fcn_update_tinyrange_bbs (fcn);
+		free (bbuf);
+	}
+	if (old_num_bbs >= r_list_length (fcn->bbs)) {
+		eprintf ("fcn_recurse failed\n");
+		return false;
 	}
-	bb->addr = addr;
+	bb = r_list_tail (fcn->bbs)->data;
```

Comment:
List can be empty?

Reply:
no longer relevant due to dcd1f66ae9ee3cdadee69419ac4a2b29098e6c0b